lib/link.js

```diff
@@ -118,7 +118,11 @@ Link.prototype.detach = function(options) {
     self.once(Link.Detached, function(info) {
       self.removeListener(Link.ErrorReceived, onError);
       if (!!info.error) return reject(info.error);
-      if (!info.closed) return reject('link not closed');
+      if (!info.closed) {
+        if (!options || options.closed) {
+          return reject('link not closed');
```

Comment:
yeah that's what this is doing:

- if the 'Detached' event that was emitting indicates that the link was not closed (!info.closed)
- then if either no options {} was supplied to the .detach(..) call, OR the options had requested closed === true, then we reject the promise because we wanted a close and we didn't get one
- otherwise we fall through to the default resolve() behaviour